base/mpfr.jl

```diff
@@ -302,18 +357,21 @@ BigFloat(x::Integer; precision::Integer=DEFAULT_PRECISION[]) =
 BigFloat(x::Integer, r::MPFRRoundingMode; precision::Integer=DEFAULT_PRECISION[]) =
     BigFloat(BigInt(x)::BigInt, r; precision=precision)
 
-BigFloat(x::Union{Bool,Int8,Int16,Int32}, r::MPFRRoundingMode=ROUNDING_MODE[]; precision::Integer=DEFAULT_PRECISION[]) =
-    BigFloat(convert(Clong, x), r; precision=precision)
-BigFloat(x::Union{UInt8,UInt16,UInt32}, r::MPFRRoundingMode=ROUNDING_MODE[]; precision::Integer=DEFAULT_PRECISION[]) =
-    BigFloat(convert(Culong, x), r; precision=precision)
-
-BigFloat(x::Union{Float16,Float32}, r::MPFRRoundingMode=ROUNDING_MODE[]; precision::Integer=DEFAULT_PRECISION[]) =
-    BigFloat(Float64(x), r; precision=precision)
+BigFloat(x::Bool, r::MPFRRoundingMode=ROUNDING_MODE[]; precision::Integer=DEFAULT_PRECISION[]) =
+     BigFloat(convert(Clong, x), r; precision=precision)
 
 function BigFloat(x::Rational, r::MPFRRoundingMode=ROUNDING_MODE[]; precision::Integer=DEFAULT_PRECISION[])
-    setprecision(BigFloat, precision) do
-        setrounding_raw(BigFloat, r) do
-            BigFloat(numerator(x))::BigFloat / BigFloat(denominator(x))::BigFloat
+    if is_bigfloat_special(x)
+        BigFloat_with_special_values(x, precision)::BigFloat
+    elseif ispow2(denominator(x))
+        z = BigFloat(numerator(x), r; precision)::BigFloat
+        z.exp -= trailing_zeros(denominator(x))
+        z
+    else
+        setprecision(BigFloat, precision) do
+            setrounding_raw(BigFloat, r) do
+                BigFloat(numerator(x))::BigFloat / BigFloat(denominator(x))::BigFloat
```

Comment:
is there not a GMP/MPFR function to divide 2 `BigInt`s?

Reply:
GMP division is integer division. MPFR does not have one for two `BigInt`s but it does have to divide a `BigFloat` by a `Clong`, `Culong` or a `BigInt` and those are already defined in `mpfr.jl`. I removed the conversion for the denominator in order to fall back to those division functions.